library/core/src/array/mod.rs

```diff
@@ -155,6 +155,28 @@ impl<T: fmt::Debug, const N: usize> fmt::Debug for [T; N] {
     }
 }
 
+// Note: the `#[rustc_skip_array_during_method_dispatch]` on `trait IntoIterator`
+// hides this implementation from explicit `.into_iter()` calls on editions < 2021,
+// so those calls will still resolve to the slice implementation, by reference.
+#[cfg(not(bootstrap))]
+#[stable(feature = "array_into_iter_impl", since = "1.53.0")]
+impl<T, const N: usize> IntoIterator for [T; N] {
+    type Item = T;
+    type IntoIter = IntoIter<T, N>;
+
+    /// Creates a consuming iterator, that is, one that moves each value out of
+    /// the array (from start to end). The array cannot be used after calling
+    /// this unless `T` implements `Copy`, so the whole array is copied.
+    ///
+    /// Arrays have special behavior when calling `.into_iter()` prior to the
+    /// 2021 edition -- see the [array] Editions section for more information.
+    ///
+    /// [array]: prim@array
```

Comment:
FWIW, I tried to anchor `#editions` directly, but I got this error:

```
error: `prim@array#editions` contains an anchor, but links to builtin types are already anchored
   --> library/core/src/array/mod.rs:174:27
    |
174 |     /// [array Editions]: prim@array#editions
    |                           ^^^^^^^^^^^^^^^^^^^ contains invalid anchor
    |
    = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
```

It doesn't even seem true that they're already anchored though... cc @jyn514

Reply:
This is https://github.com/rust-lang/rust/issues/83083.

Reply:
Aha, thanks!